File: app/src/main/java/mb00/android/codehub/ui/user/viewmodel/UserPulseViewModel.java
package mb00.android.codehub.ui.user.viewmodel;

import mb00.android.codehub.ui.user.interactor.UserInteractor;
import mb00.android.codehub.ui.user.router.UserRouter;


public class UserPulseViewModel extends UserViewModel {

    private final UserInteractor interactor;
    private final UserRouter router;

    public UserPulseViewModel(UserInteractor interactor, UserRouter router) {
        super(interactor, router);
        this.interactor = interactor;
        this.router = router;
    }

    /*public void getUserPulse(String header, String user) {
        apiCallManager.loadUserPulse(header, user);
    }*/

}

File: app/src/main/java/mb00/android/codehub/ui/user/viewmodel/UserPulseViewModel.kt
package mb00.android.codehub.ui.user.viewmodel

import io.reactivex.Single
import mb00.android.codehub.api.model.Pulse
import mb00.android.codehub.ui.user.interactor.UserInteractor
import mb00.android.codehub.ui.user.router.UserRouter


class UserPulseViewModel(private val interactor: UserInteractor, private val router: UserRouter) : UserViewModel(interactor, router) {

    fun loadUserPulse(header: String, user: String): Single<List<Pulse>> {
        return interactor.loadUserPulse(header, user)
    }

}
